tools/clang/lib/Parse/ParseExpr.cpp

```diff
@@ -1094,7 +1094,7 @@ ExprResult Parser::ParseCastExpression(bool isUnaryExpression,
   case tok::kw___real:     // unary-expression: '__real' cast-expression [GNU]
   case tok::kw___imag: {   // unary-expression: '__imag' cast-expression [GNU]
     // HLSL Change Starts
-    if (getLangOpts().HLSL && Tok.getKind() == tok::kw___real || Tok.getKind() == tok::kw___imag) {
+    if ((getLangOpts().HLSL && Tok.getKind() == tok::kw___real) || Tok.getKind() == tok::kw___imag) {
```

Comment:
I'm wondering whether we should group `kw__real` and `kw__imag` together here.

Reply:
Good catch. I pretty readily grouped this one according to existing behavior, but all things being equal, that's probably closer to the intent. Because && has greater precedence than ||, this is the change that doesn't alter the behavior. 

However, LangOpts::HLSL is always true because lang opts are statically assigned in the current build variants. So it wouldn't change the behavior regardless of where the parens go.

 I'm tempted to move this code around so the switch cases remove the need to check the kinds again, but that would involve a fallthrough, which a lot of programmers tend to look unfavorably upon.